internal/backend/rest/rest.go

```diff
@@ -349,3 +353,48 @@ func (b *restBackend) Close() error {
 	// same function.
 	return nil
 }
+
+// Writable implements restic.Writabler for the REST backend and returns
+// whether the backend can be written to or must be considered as read-only.
+//
+// The methods probes the backend by writing to a temporary file and deleting
+// it wright away. If that fails for any reason, the backend is considered to
+// be read-only.
+func (b *restBackend) Writable() bool {
+	debug.Log("Writable()")
+
+	const probeFilename = ".resticWriteCheck"
```

Comment:
This can fail when two instances of restic create this file simultaneously. Or restic failed, but left this file behind. The name must be unpredictable.